protobuf-serializer/src/main/java/io/confluent/kafka/formatter/protobuf/ProtobufMessageReader.java

```diff
@@ -0,0 +1,239 @@
+  private String keySubject = null;
+  private String valueSubject = null;
+  private Serializer keySerializer;
+
+  /**
+   * Constructor needed by kafka console producer.
+   */
+  public ProtobufMessageReader() {
+  }
+
+  /**
+   * For testing only.
+   */
+  ProtobufMessageReader(
+      SchemaRegistryClient schemaRegistryClient,
+      ProtobufSchema keySchema,
+      ProtobufSchema valueSchema,
+      String topic,
+      boolean parseKey,
+      BufferedReader reader,
+      boolean autoRegister
+  ) {
+    this.schemaRegistry = schemaRegistryClient;
+    this.keySchema = keySchema;
+    this.valueSchema = valueSchema;
+    this.topic = topic;
+    this.keySubject = topic + "-key";
+    this.valueSubject = topic + "-value";
+    this.parseKey = parseKey;
+    this.reader = reader;
+    this.autoRegisterSchema = autoRegister;
+  }
+
+  @Override
+  public void init(java.io.InputStream inputStream, Properties props) {
+    topic = props.getProperty("topic");
+    if (props.containsKey("parse.key")) {
+      parseKey = props.getProperty("parse.key").trim().toLowerCase().equals("true");
+    }
+    if (props.containsKey("key.separator")) {
+      keySeparator = props.getProperty("key.separator");
+    }
+    if (props.containsKey("ignore.error")) {
+      ignoreError = props.getProperty("ignore.error").trim().toLowerCase().equals("true");
+    }
+    reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
+    String url = props.getProperty(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG);
+    if (url == null) {
+      throw new ConfigException("Missing schema registry url!");
+    }
+
+    Map<String, Object> originals = getPropertiesMap(props);
+
+    schemaRegistry = new CachedSchemaRegistryClient(Collections.singletonList(url),
+        AbstractKafkaSchemaSerDeConfig.MAX_SCHEMAS_PER_SUBJECT_DEFAULT,
+        Collections.singletonList(new ProtobufSchemaProvider()),
+        originals
+    );
+    if (!props.containsKey("value.schema")) {
+      throw new ConfigException("Must provide the Protobuf schema string in value.schema");
+    }
+    String valueSchemaString = props.getProperty("value.schema");
+    valueSchema = new ProtobufSchema(valueSchemaString);
+
+    keySerializer = getKeySerializer(props);
+
+    if (needsKeySchema()) {
+      if (!props.containsKey("key.schema")) {
+        throw new ConfigException("Must provide the Protobuf schema string in key.schema");
+      }
+      String keySchemaString = props.getProperty("key.schema");
+      keySchema = new ProtobufSchema(keySchemaString);
+    }
+    keySubject = topic + "-key";
+    valueSubject = topic + "-value";
+    if (props.containsKey("auto.register")) {
+      this.autoRegisterSchema = Boolean.parseBoolean(props.getProperty("auto.register").trim());
+    } else {
+      this.autoRegisterSchema = true;
+    }
+  }
+
+  private Serializer getKeySerializer(Properties props) throws ConfigException {
```

Comment:
Nit: do we need the throws declaration here ? ConfigException is a RuntimeException.

Reply:
It's not strictly necessary to declare the exception; however, this is also in the AvroMessageReader so I think I'll leave it for now